lib/util/mapObject.ts

```diff
@@ -0,0 +1,14 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+export interface MapObject<T> {
+  [name: string]: T
+}
+
+export function objectMap<T>(src: MapObject<T>, f: (v: T, name: string) => T): MapObject<T> {
```

Comment:
>function [](start = 7, length = 8)

dont we want to have arrow functions wherever we can ? or afraid to break any behavior ?